include/fiction/io/read_fgl_layout.hpp

```diff
@@ -0,0 +1,813 @@
+//
+// Created by simon on 25.09.23.
+//
+
+#ifndef FICTION_READ_FGL_LAYOUT_HPP
+#define FICTION_READ_FGL_LAYOUT_HPP
+
+#include "fiction/layouts/cartesian_layout.hpp"
+#include "fiction/layouts/clocked_layout.hpp"
+#include "fiction/layouts/gate_level_layout.hpp"
+#include "fiction/layouts/tile_based_layout.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/utils/name_utils.hpp"
+
+#include <kitty/constructors.hpp>
+#include <tinyxml2.h>
+
+#include <algorithm>
+#include <cctype>
+#include <cstdlib>
+#include <exception>
+#include <fstream>
+#include <istream>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
```

Comment:
Please only use warning-suppressing statements around the problematic code blocks and not around an entire file as they might suppress too many potentially justified and useful warnings.